### PR TITLE
Fixes #16782 - reset dhcp record cache after mac change

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
test error unrelated

File: app/views/nic/_base_form.html.erb [hunk 1]
```diff
@@ -4,7 +4,7 @@
                     :size => "col-md-8", :label_size => "col-md-3" %>
 
 <%= text_f f, :mac,
-              :help_inline => popover('', _("Media access control address for this interface. Format must be six groups of two hexadecimal digits separated by colons (:), e.g. 00:11:22:33:44:55"), :rel => 'popover-modal'),
+	:help_inline => popover('', _("Media access control address for this interface. Format must be six groups of two hexadecimal digits separated by colons (:), e.g. 00:11:22:33:44:55. Note that if the host is deployed on compute resource that generates MAC address such as VMware, oVirt, libvirt, this field will be set automatically when the resource is created and will override any custom value that was provided."), :rel => 'popover-modal'),
```
Comment (timogoebel) [inline app/views/nic/_base_form.html.erb]: Nitpick: Could you please fix the indent here?

**Review by timogoebel — APPROVED** [review]
Just one nitpick. The rest looks good to me. @lzap : Your turn! ;-)

**Comment by ares** [comment]
Thanks Timo, I updated the indentation to be the same as few lines below. @lzap anything to add?

**Comment by lzap** [comment]
Just a nitpick, this looks too long: "Note that if the host is deployed on compute resource that generates MAC address such as VMware, oVirt, libvirt, this field will be set automatically when the resource is created and will override any custom value that was provided."

How about something shorter like "Most virtualization compute resources (e.g. VMware, oVirt, libvirt) will overwrite the provided value with new random MAC address."

**Review by lzap — APPROVED** [review]

**Comment by timogoebel** [comment]
> "Most virtualization compute resources (e.g. VMware, oVirt, libvirt) will overwrite the provided value with new random MAC address."

How about?

"Most virtualization compute resources (e.g. VMware, oVirt, libvirt) will provide this value."

**Comment by lzap** [comment]
Well I am fine with that, but I'd stress the fact that the custom value will be overwritten (lost really).

**Comment by timogoebel** [comment]
Ok. I'm fine with any suggestion.
@ares : Could you please shorten the sentence? Thanks.

**Comment by ares** [comment]
I shortened the sentence, please take another look.

**Comment by lzap** [comment]
Thanks @ares!

**Pull request merged**